src/machine/machine_stm32_moder_gpio.go

```diff
@@ -0,0 +1,142 @@
+// +build stm32,!stm32f103xx
+
+package machine
+
+// GPIO for the stm32 families except the stm32f1xx which uses a simpler but
+//  less flexible mechanism. Extend the +build directive above to exclude other
+//  models in the stm32f1xx series as necessary
+
+const (
+	// Mode Flag
+	PinOutput        PinMode = 0
+	PinInput         PinMode = PinInputFloating
+	PinInputFloating PinMode = 1
+	PinInputPulldown PinMode = 2
+	PinInputPullup   PinMode = 3
+
+	// for UART
+	PinModeUartTX PinMode = 4
+	PinModeUartRX PinMode = 5
```

Comment:
The [usual Go convention](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) is to keep abbreviations in uppercase:

> Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. For example, "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url". As an example: ServeHTTP not ServeHttp. For identifiers with multiple initialized "words", use for example "xmlHTTPRequest" or "XMLHTTPRequest".

However, in this case I'm not so sure as multiple abbreviations follow each other. [More discussion here](https://stackoverflow.com/a/39830270/559350). @deadprogram any opinion on this?

Reply:
how about `PinModeUARTTX` and `PinModeUARTRX`? I can fix up the SPI and I2C constants too.

Reply:
Actually there pins are pretty standard. Periph.io define these pins in a standalone package and they are shared among all hardwares, for example: https://godoc.org/periph.io/x/periph/conn/uart.

Reply:
that's a pretty cool looking library! Good consideration for all the different types of pin usage likely to happen. I don't know which solution would be better:
- update the existing definitions in the one file, really just for the benefit of `stm32` users
- move the definitions to e.g. `uart.go`, `spi.go`, `i2c.go` etc in a similar way to the Periph.io library for other board/chip users to leverage in a common way
- import Periph.io and use its definitions directly

Thoughts? I'll go down the first path for now, as it's the minimal change, but this seems like a good thing to standardize for the tinygo platform.

Reply:
I agree with you on the first path to make this MR more feasible to be merged.

Reply:
Pin modes fall roughly in two groups:

* Modes that are standard across chips and vendors, such as `PinInput`, `PinOutput`, `PinInputPullup`, etc. These modes are common across many chips and therefore should have a consistent name to make it easier to write portable code. Of course, when a given mode is not available on a chip, it should not be defined to get a clear compiler error.
* Modes that are chip (family/vendor) specific. Most chips have these, for example the SAM D family of chips has the "SERCOM" and "SERCOMAlt" modes used for serial protocols (SPI, I2C, ...). The UART Tx mode also falls in this category. They are inherently unportable and mostly used inside the machine package anyway so the exact name is less critical than the standard pin modes.

Long story short: the constants you have now look fine by me.